src/compile/data.ts

```diff
@@ -667,16 +690,63 @@ export namespace summary {
     return [];
   }
 
+  function mergeMeasures(parentMeasures, childMeasures) {
+    for (var field in childMeasures) {
+      if (childMeasures.hasOwnProperty(field)) {
+        // when we merge a measure, we either have to add an aggregation operator or even a new field
+        var ops = childMeasures[field];
+        for (var op in ops) {
+          if (ops.hasOwnProperty(op)) {
+            if (field in parentMeasures) {
+              // add operator to existing measure field
+              parentMeasures[field][op] = true;
+            } else {
+              parentMeasures[field] = {op: true};
+            }
+          }
+        }
+      }
+    }
+  }
+
   export function parseLayer(model: LayerModel): SummaryComponent[] {
-    // we create new data sources for each summary in a layer
-    return [];
+    // indx by the fields we are grouping by
+    let summaries = {} as Dict<SummaryComponent>;
+
+    // combine summaries for children that don't have a source
+    model.children().forEach((child) => {
+      const childDataComponent = child.component.data;
+      if (!childDataComponent.source && childDataComponent.summary) {
+        // let's try to merge the summaries if we can
+        childDataComponent.summary.forEach((summary) => {
+          // the key is a key of the dimensions
+          // we use it to find out whether we have a summary that uses the same group by fields
+          const key = stringSetHash(summary.dimensions);
+          if (key in summaries) {
+            // yes, there is a summary hat we need to merge into
+            // we know that the dimensions are the same so we only need to merge the measures
+            mergeMeasures(summaries[key].measures, summary.measures);
+          } else {
+            // give the summary a new name
+            summary.name = model.dataName(SUMMARY) + '_' + keys(summaries).length;
```

Comment:
Good point. I changed the way summaries are generated and now it would make more sense to use the fields (+the source name?). But I don't want to worry about all the edge cases (explicit values, ...)